Formula/b/bozohttpd.rb

```diff
@@ -0,0 +1,49 @@
+class Bozohttpd < Formula
+  desc "Small and secure http version 1.1 server"
+  homepage "http://www.eterna.com.au/bozohttpd/"
+  url "http://www.eterna.com.au/bozohttpd/bozohttpd-20220517.tar.bz2"
+  sha256 "9bfd0942a0876e5529b0d962ddbcf50473bcf84cf5e4447043e4a0f4ea65597a"
+  license "BSD-2-Clause"
+
+  depends_on "pkg-config" => :build
+  depends_on "rtmpdump" => [:build, :test]
+  depends_on "lua"
+  depends_on "openssl@3"
+
+  uses_from_macos "curl" => [:build, :test]
+
+  def install
+    cflags = [
+      "-O",
+      "-D_LARGEFILE_SOURCE",
+      "-D_FILE_OFFSET_BITS=64",
+      "_GNU_SOURCE",
+      "-D_DEFAULT_SOURCE",
+      "-DNO_BLOCKLIST_SUPPORT",
+      "-DDO_HTPASSWD",
+    ]
```

Comment:
Adding to what @fxcoudert said: how did you figure out/decide what was required/not here?

Reply:
bozohttpd is a NetBSD package. It has a very specific NetBSD Makefile and a 'cross-platform' Makefile (Makefile.boot). I took the flags from the `Makefile.boot` file. I didn't check the source code how and why these flags (first 5) are used. They look internal. I'm trying to get more insight into these from the authors.
The two last flags are for specific features. Because `blocklistd` is NetBSD specific I did omit that one. Because basic auth is disabled by default, I added that one specifically as it seems useful to have.

Reply:
Can you link to these various resources you used to make this decisions and turn more of this explanation into comments in the Ruby formula source? Thanks!

Reply:
I contacted the bozohttpd author to figure out what the flags mean:
  * -O is just a good default for optimisation.
  *  _LARGEFILE_SOURCE, _FILE_OFFSET_BITS=64 makes 32-bit linux hosts capable of serving files larger then 4GiB.
  * _DEFAULT_SOURCE, _GNU_SOURCE  enables some things hidden by default by glibc -- this also is for linux

"For macos i would expect _only_ some optimization is what
you need, i.e., depending on builder options '-g', '-O',
'-O2', etc. (maybe -g and -O together, too.)  the choice
there would be up to the packaging."

Reply:
I pushed some changes removing the flags we don't need.

Reply:
Optimisation is managed by brew, so that can be removed.